unittests/AST/ASTImporterTest.cpp

```diff
@@ -3598,6 +3598,45 @@ TEST_P(ASTImporterTestBase, ImportUnnamedStructsWithRecursingField) {
       R1, recordDecl(has(fieldDecl(hasName("next"))))));
 }
 
+TEST_P(ASTImporterTestBase, ImportUnnamedFieldsInCorrectOrder) {
```

Comment:
I just have run this test on the tip of Ericsson/ctu-clang6 and it passed. It should fail, shouldn't it?
Or is it not related to the change?

I'd expect such a test to this change where you explicitly set a Decl to have external lexical storage.

Reply:
The test is added to have a test for this, and it is related (but does not fail). The actual case is not easy (?) to reproduce in a test: The fields should exist in external AST source but not loaded yet. (Or is there a quick way to test this case?)

Reply:
What you can do is to write an ast-merge lit test, where we import from a PCH. When we load the AST from the PCH (but before we start the import), then the DeclContext must have external lexical storage.